### ISSUE TITLE
Catalogs pulling wrong season

### ISSUE THREAD

**Issue by TheGenuineGT** [Jenrykster/animeo#3]
First, thank you for working this project. It is immensely appreciated and I wish I was savvy enough in node.js to help.

I noticed you just updated everything yesterday, and for the most part everything works great. My catalogs are accurate for currently watching, dropped, completed, etc.

The only issue I'm seeing currently is the catalogs appear to be pulling the incorrect season. Example; Jujutsu Kaisen Season 2 is in my currently watching, listed as Jujutsu Kaisen Season 2. However when clicking the show to pull the episodes, it then pulls season one. This issue exists for some anime but not all, jujutsu kaiesen and faraway paladin for example pull the wrong season. But Tokyo revengers and spy x family pull the correct season.

I am also unable to add the shows from the catalog to my library in stremio, I assumed because it was now pulling from Anilist instead of kitsu but looking through your code it doesn't appear to be the case.

Added images for reference

![catalog works](https://github.com/Jenrykster/animeo/assets/145689387/2ad47dfa-1fb2-4c97-b0c9-53334f77eff8)
![wrong season](https://github.com/Jenrykster/animeo/assets/145689387/d371c0f1-86cf-402c-98e0-a98aa38e91c3)

**Comment by Jenrykster** [comment]
Hey, thanks for the kind words and testing the addon. 

It should still be using kitsu to get the season and episodes info, I'll look into it.

**Comment by TheGenuineGT** [comment]
Small update. I am unsure if things worked themselves out, or if adding "Cyberflix Catalog" and "The Movie Database Addon" made the difference. But now It appears that the Anilist watchlists are pulling the correct seasons for all of the anime on my lists. It does still show as a separate show than the Kitsu version added to my Stremio library, with show's not already added to my library able to be added. But there is a bug that doesn't show on the Anime page that it was added.

~~I will test later if watching a new episode from the anime on my watchlist still updates my anilist profile.~~
It does indeed update my anilist profile. Again, appreciate you working on this!

**Comment by Jenrykster** [comment]
Glad you like it.

So, about the original bug report... It seems stremio changes which addon provides metadata for a series based on some rules (which I haven't figured yet). As a test I tried providing metadata from anylist to always get the correct entry but to get it working I had to uninstall kitsu (and then install it again) so I'm not entirely sure if stremio is just using cinemeta sometimes (which uses a different season system).

But anyways, I was planning to update animeo with this new metadata handler but It seems like anilist doesnt provide metadata from specific episodes, making streaming impossible. I even tried merging kitsu episode info with anilist data but the result wasn't that good so I decided to drop it.

So, in summary, I won't fix this for now and **if anyone is having the same problem they should try uninstalling the catalog addons and then reinstalling then in a different order** (or installing the same one you did).

also, sorry for taking so long to reply, was busy with some other projects.

**Comment by TheGenuineGT** [comment]
No worries brother, thanks again for working on this. I know it's a side project for you. I'd be more than happy to buy you a coffee if you drop a link in the readme \m/

**Comment by Jenrykster** [comment]
> No worries brother, thanks again for working on this. I know it's a side project for you. I'd be more than happy to buy you a coffee if you drop a link in the readme \m/

Well, sure I'd appreciate it ;)
